component/org.wso2.carbon.identity.api.otp.service.emailotp/src/main/java/org/wso2/carbon/identity/api/otp/service/emailotp/impl/EmailotpApiServiceImpl.java

```diff
@@ -107,4 +110,42 @@ public Response emailotpValidatePost(OTPValidationRequest otpValidationRequest)
             return EndpointUtils.handleUnexpectedServerError(e, log);
         }
     }
+
+    /**
+     * This method is implemented from org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi
+     * to verify OTP without invalidate.
+     *
+     * @param otpVerificationRequest Otp verification request object.
+     * @return Response
+     */
+    @Override
+    public Response emailotpVerifyPost(OTPVerificationRequest otpVerificationRequest) {
+
+        String transactionId = StringUtils.trim(otpVerificationRequest.getTransactionId());
+        String userId = StringUtils.trim(otpVerificationRequest.getUserId());
+        String emailOtp = StringUtils.trim(otpVerificationRequest.getEmailOtp());
+        try {
+            ValidationResponseDTO responseDTO = EndpointUtils.getEmailOTPService().verifyEmailOTP(
+                    transactionId, userId, emailOtp);
+            FailureReasonDTO failureReasonDTO = responseDTO.getFailureReason();
+            OTPVerificationFailureReason failureReason = null;
+            if (failureReasonDTO != null) {
+                failureReason = new OTPVerificationFailureReason()
+                        .code(failureReasonDTO.getCode())
+                        .message(failureReasonDTO.getMessage())
+                        .description(failureReasonDTO.getDescription());
+            }
+            OTPVerificationResponse response = new OTPVerificationResponse()
+                    .isValid(responseDTO.isValid())
+                    .userId(responseDTO.getUserId())
+                    .failureReason(failureReason);
```

Comment:
why are we setting a failure reason here?

Reply:
It should be in the response if the responseDTO.isValid() == false